doc/source/programs/gdal_vector_sql.rst

```diff
@@ -0,0 +1,105 @@
+.. _gdal_vector_sql_subcommand:
+
+================================================================================
+"gdal vector sql" sub-command
+================================================================================
+
+.. versionadded:: 3.11
+
+.. only:: html
+
+    Apply SQL statement(s) to a dataset
+
+.. Index:: gdal vector sql
+
+Synopsis
+--------
+
+.. code-block::
+
+    Usage: gdal vector sql [OPTIONS] <INPUT> <OUTPUT> <statement>|@<filename>
+
+    Apply SQL statement(s) to a dataset.
+
+    Positional arguments:
+      -i, --input <INPUT>                                  Input vector dataset [required]
+      -o, --output <OUTPUT>                                Output vector dataset [required]
+      --sql <statement>|@<filename>                        SQL statement(s) [may be repeated] [required]
+
```

Comment:
The number of `--sql`s must match the number of the layers defined later with `--output-layer`, right?

Reply:
yes, I've added doc for --output-layer mentionning this